packages/core/src/chain.ts

```diff
@@ -20,15 +21,16 @@ async function setChain(options: {
 
   const { wallets, chains } = state.get()
   const { chainId, chainNamespace = 'evm', wallet: walletToSet } = options
+  const chainIdHex = typeof chainId === 'number' ? decimalToHex(chainId) : chainId
 
   // validate that chainId has been added to chains
   const chain = chains.find(
-    ({ namespace, id }) => namespace === chainNamespace && id === chainId
+    ({ namespace, id }) => namespace === chainNamespace && id === chainIdHex
   )
 
   if (!chain) {
     throw new Error(
-      `Chain with chainId: ${chainId} and chainNamespace: ${chainNamespace} has not been set and must be added when Onboard is initialized.`
+      `Chain with chainId: ${chainIdHex} and chainNamespace: ${chainNamespace} has not been set and must be added when Onboard is initialized.`
```

Comment:
```suggestion
      `Chain with chainId: ${chainId} and chainNamespace: ${chainNamespace} has not been set and must be added when Onboard is initialized.`
```
We can use the original chainId for this message to prevent confusion, as the dev passes in a decimal and the error message indicates that they passed in a hex.